swarm/storage/types.go

```diff
@@ -241,9 +242,44 @@ The ChunkStore interface is implemented by :
 type ChunkStore interface {
 	Put(*Chunk) // effectively there is no error even if there is an error
 	Get(Key) (*Chunk, error)
+	Validate(*Key, []byte) bool
```

Comment:
not sure if this interface is right. The validation does not depend on the store.
It depends on the chunk.
We need to add a sequence of validators (content address, mutable resource) and use those.
Maybe in the network when we validate we could use an extra field in the chunk messages to indicate the chunk type.  otherwise just fall back to the next if the previous validator fails

Reply:
@zelig It's not quite that simple. The Resource chunk validation is not necessarily uniform.

How would fallback work here? If my Resource A validation fails because the signature is not valid according to the validation criteria in resource A, then why would we need to check anything else?

The "fallback" could be that if no chunk type is specified, it will be processed by a default validator (which in most cases would be the content address validator).